examples/tests/valid/store.w

```diff
@@ -1,3 +1,6 @@
+bring "./subdir/empty.w" as file3;
```

Comment:
I got to say that I really dont like that we are requiring the `.w` here. @MarkMcCulloh i remember you were very opinionated about it. Can you remind me why?

Reply:
Omitting the file extension has caused many issues in the Node ecosystem:

- Runtime code has to do expensive look ups. For example, when doing `require("./subdir/empty")`, Node needs to check:
  - Does `./subdir/empty.js` exist?
  - Does `./subdir/empty/index.js` exist?
  - Not sure if it also checks for `.json` files...
- Nowadays, there's not only `.js` files. There are:
  - TypeScript files `.ts`
  - TSX and JSX `.tsx, .jsx`
  - CommonJS files `.cjs`
  - ESM files `.mjs`
  - You can configure a bundler to load any file extension with the logic you need (`.vue`, `.svelte`

Mix all of those and we get the mess we have now. This has been corrected in ESM now, since providing the full filename is necessary now.

I think we should follow the modern trend and force the full filename since the beginning. I believe we will end up being able to bring/import more file formats.

Reply:
^+1, basically we should learn from past mistakes. 
Considering the file extension is one letter, the price to pay isn't even too bad

Reply:
I think currently omitting the extension would mix this up with the JSII bringing. Of course we can have some heuristics about this, but still. I suggest for now to require it. Removing it later as a feature can also work.